engine/src/flutter/lib/web_ui/lib/src/engine/semantics/scrollable.dart

```diff
@@ -183,64 +172,51 @@ class SemanticScrollable extends SemanticRole {
+  /// Updates the scrollable state of the semantics object.
+  void _updateScrollableState() {
     // This value is arbitrary.
-    const int canonicalNeutralScrollPosition = 10;
     final ui.Rect? rect = semanticsObject.rect;
     if (rect == null) {
       printWarning('Warning! the rect attribute of semanticsObject is null');
       return;
     }
     if (semanticsObject.isVerticalScrollContainer) {
-      // Place the _scrollOverflowElement at the end of the content and
-      // make sure that when we neutralize the scrolling position,
-      // it doesn't scroll into the visible area.
-      final int verticalOffset = rect.height.ceil() + canonicalNeutralScrollPosition;
+      // Place the _scrollOverflowElement at the beginning of the content
+      // and size it based on the total scroll extent so the browser
+      // knows how much scrollable content there is.
+      final double? scrollExtentMax = semanticsObject.scrollExtentMax;
+      final double? scrollExtentMin = semanticsObject.scrollExtentMin;
+      assert(scrollExtentMax != null);
+      assert(scrollExtentMin != null);
+      final double scrollExtentTotal = scrollExtentMax! - scrollExtentMin! + rect.height.round();
       _scrollOverflowElement.style
-        ..transform = 'translate(0px,${verticalOffset}px)'
         ..width = '${rect.width.round()}px'
-        ..height = '${canonicalNeutralScrollPosition}px';
-
-      element.scrollTop = canonicalNeutralScrollPosition.toDouble();
-      // Read back because the effective value depends on the amount of content.
-      _effectiveNeutralScrollPosition = element.scrollTop.toInt();
+        ..height = '${scrollExtentTotal}px';
       semanticsObject
-        ..verticalScrollAdjustment = _effectiveNeutralScrollPosition.toDouble()
+        ..verticalScrollAdjustment = element.scrollTop
         ..horizontalScrollAdjustment = 0.0;
     } else if (semanticsObject.isHorizontalScrollContainer) {
-      // Place the _scrollOverflowElement at the end of the content and
-      // make sure that when we neutralize the scrolling position,
-      // it doesn't scroll into the visible area.
-      final int horizontalOffset = rect.width.ceil() + canonicalNeutralScrollPosition;
+      // Place the _scrollOverflowElement at the beginning of the content
+      // and size it based on the total scroll extent so the browser
+      // knows how much scrollable content there is.
+      final double? scrollExtentMax = semanticsObject.scrollExtentMax;
+      final double? scrollExtentMin = semanticsObject.scrollExtentMin;
+      assert(scrollExtentMax != null);
+      assert(scrollExtentMin != null);
+      final double scrollExtentTotal = scrollExtentMax! - scrollExtentMin! + rect.width.round();
       _scrollOverflowElement.style
-        ..transform = 'translate(${horizontalOffset}px,0px)'
-        ..width = '${canonicalNeutralScrollPosition}px'
+        ..width = '${scrollExtentTotal}px'
         ..height = '${rect.height.round()}px';
```

Comment:
Ditto re height value, can it be something static?

Reply:
https://github.com/flutter/flutter/pull/165589#discussion_r2032001147